crates/services/txpool_v2/src/pool.rs

```diff
@@ -0,0 +1,205 @@
+                self.config.black_list.check_blacklisting(&tx)?;
+                Self::check_blob_does_not_exist(&tx, &latest_view)?;
+                self.storage.validate_inputs(
+                    &tx,
+                    &latest_view,
+                    self.config.utxo_validation,
+                )?;
+                let collisions = self
+                    .collision_manager
+                    .collect_colliding_transactions(&tx, &self.storage)?;
+                let dependencies = self.storage.collect_transaction_dependencies(&tx)?;
+                let has_dependencies = !dependencies.is_empty();
+                let (storage_id, removed_transactions) = self.storage.store_transaction(
+                    tx,
+                    dependencies,
+                    collisions.colliding_txs(),
+                )?;
+                #[cfg(test)]
+                {
+                    self.tx_id_to_storage_id.insert(tx_id, storage_id);
+                }
+                // No dependencies directly in the graph and the sorted transactions
+                if !has_dependencies {
+                    self.selection_algorithm
+                        .new_executable_transactions(vec![storage_id], &self.storage)?;
+                }
+                self.update_components_and_caches_on_removal(
+                    removed_transactions.iter(),
+                )?;
+                let tx = Storage::get(&self.storage, &storage_id)?;
+                self.collision_manager
+                    .on_stored_transaction(&tx.transaction, storage_id)?;
+                Ok(removed_transactions)
+            })
+            .collect())
+    }
+
+    // TODO: Use block space also (https://github.com/FuelLabs/fuel-core/issues/2133)
+    /// Extract transactions for a block.
+    /// Returns a list of transactions that were selected for the block
+    /// based on the constraints given in the configuration and the selection algorithm used.
+    pub fn extract_transactions_for_block(
+        &mut self,
+    ) -> Result<Vec<PoolTransaction>, Error> {
+        let extracted_transactions = self
+            .selection_algorithm
+            .gather_best_txs(
+                Constraints {
+                    max_gas: self.config.max_block_gas,
+                },
+                &self.storage,
+            )?
+            .into_iter()
+            .map(|storage_id| {
+                let storage_data = self
+                    .storage
+                    .remove_transaction_without_dependencies(storage_id)?;
+                Ok(storage_data.transaction)
+            })
+            .collect::<Result<Vec<PoolTransaction>, Error>>()?;
+        self.update_components_and_caches_on_removal(extracted_transactions.iter())?;
+        Ok(extracted_transactions)
+    }
+
+    /// Prune transactions from the pool.
+    pub fn prune(&mut self) -> Result<Vec<PoolTransaction>, Error> {
+        Ok(vec![])
+    }
```

Comment:
If we don't use it, maybe it is better to remove for now

Reply:
Ok, done.